tests/integration/v3_watch_restore_test.go

```diff
@@ -90,19 +94,38 @@ func TestV3WatchRestoreSnapshotUnsync(t *testing.T) {
 	kvc := integration.ToGRPC(clus.Client(1)).KV
 
 	// to trigger snapshot from the leader to the stopped follower
-	for i := 0; i < 15; i++ {
+	//
+	// NOTE: In 3 members cluster, after initial lead has been elected,
+	// there are 3xConfChange + 3xMemberAttrSet + 1xClusterVersionSet logs.
+	// So member 0 has index 8 in raft log before network partition. We
+	// need to trigger snapshot at least twice.
+	//
+	// SnapshotCount: 10, SnapshotCatchUpEntries: 5
+	// T1: L(snapshot-index: 11, compacted-index:  6), F_m0(index:8)
+	// T2: L(snapshot-index: 22, compacted-index: 17), F_m0(index:8, out of date)
+	// T3: L(snapshot-index: 33, compacted-index: 28), F_mo(index:8, out of date)
+	//
+	// Since the snapshot is handled in GoAttach, we need to trigger
+	// snapshot 3 times and then leader starts to send snapshot after
+	// the leader realizes the hint index from member 0 is out of date.
+	for i := 0; i < 35; i++ {
 		_, err := kvc.Put(context.TODO(), &pb.PutRequest{Key: []byte("foo"), Value: []byte("bar")})
 		if err != nil {
 			t.Errorf("#%d: couldn't put key (%v)", i, err)
 		}
 	}
+	// The 33 is latest snapshot index.
+	ensureTriggerSnapshot(t, clus.Members[initialLead], 33, 5*time.Second)
```

Comment:
ideally this should wait for leader to compact index 8, but there is no way to check for that.
Writing snapshot files isn't enough to trigger snapshot send. For example if you set SnapshotCatchUpEntries to smth high, new version of the test will still fail. Maybe you can add a comment that implicitly it's also waiting for compaction.

Reply:
I switch to a solution which monitors testing log to ensure cluster has compacted the raft log. please take a look. Thanks